diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ad8a39..2919236 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,18 +36,56 @@ IF(SSPLIT_PREFER_STATIC_COMPILE)
   ENDIF(WIN32)
 ENDIF(SSPLIT_PREFER_STATIC_COMPILE)
 
+
+find_package(pcre2 CONFIG REQUIRED)
+
 include(GNUInstallDirs)
-include(FindPCRE2)
-if(PCRE2_FOUND)
-  include_directories(${PCRE2_INCLUDE_DIRS})
-  set(EXT_LIBS ${EXT_LIBS} ${PCRE2_LIBRARIES})
+include(CMakePackageConfigHelpers)
+if(pcre2_FOUND)
+    # include_directories(${PCRE2_INCLUDE_DIRS})
+  set(EXT_LIBS ${EXT_LIBS} PCRE2::8BIT PCRE2::16BIT PCRE2::32BIT PCRE2::POSIX)
   message("   Using these libraries: ${EXT_LIBS}")
-else(PCRE2_FOUND)
+else(pcre2_FOUND)
   message(SEND_ERROR "Cannot find pcre2 libraries. Terminating.")
-endif(PCRE2_FOUND)
+endif(pcre2_FOUND)
 
 add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/src)
 message("Added subdirectory ${CMAKE_CURRENT_SOURCE_DIR}/src")
 if(SSPLIT_USE_INTERNAL_PCRE2)
   add_dependencies(ssplit pcre2) # this needs to happen after adding src
 endif(SSPLIT_USE_INTERNAL_PCRE2)
+
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/src/ssplit
+    DESTINATION "include"
+    FILES_MATCHING
+      PATTERN "*h"
+)
+
+install(TARGETS ssplit
+    EXPORT ssplitTargets
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+
+install(EXPORT ssplitTargets 
+    NAMESPACE ssplit:: 
+    DESTINATION share/ssplit 
+    FILE ssplitTargets.cmake)
+
+configure_package_config_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/ssplitConfig.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/ssplitConfig.cmake"
+    INSTALL_DESTINATION share/ssplit)
+
+write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/ssplitConfigVersion.cmake"
+    VERSION 1.0.0
+    COMPATIBILITY SameMajorVersion
+)
+
+install(FILES
+    "${CMAKE_CURRENT_BINARY_DIR}/ssplitConfig.cmake"
+    "${CMAKE_CURRENT_BINARY_DIR}/ssplitConfigVersion.cmake"
+    DESTINATION share/ssplit
+)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 60e652f..e9fb430 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,10 @@
-include_directories(ssplit)
+# include_directories(ssplit)
 add_library(ssplit STATIC ssplit/ssplit.cpp ssplit/regex.cpp)
+target_include_directories(ssplit 
+    INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/ssplit>
+    $<INSTALL_INTERFACE:include/ssplit>)
+
 target_link_libraries(ssplit ${EXT_LIBS})
 
 set_target_properties(ssplit PROPERTIES
diff --git a/ssplitConfig.cmake.in b/ssplitConfig.cmake.in
new file mode 100644
index 0000000..d64d7c6
--- /dev/null
+++ b/ssplitConfig.cmake.in
@@ -0,0 +1,7 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+find_dependency(pcre2 CONFIG)
+
+include("${CMAKE_CURRENT_LIST_DIR}/ssplitTargets.cmake")
+check_required_components(ssplit)
