diff --git a/src/speechPlayer/include/speechPlayer.h b/src/speechPlayer/include/speechPlayer.h
index 2f5a9b69..3d7a5d8d 100644
--- a/src/speechPlayer/include/speechPlayer.h
+++ b/src/speechPlayer/include/speechPlayer.h
@@ -57,7 +57,7 @@ typedef void* speechPlayer_handle_t;
 
 speechPlayer_handle_t speechPlayer_initialize(int sampleRate);
 void speechPlayer_queueFrame(speechPlayer_handle_t playerHandle, speechPlayer_frame_t* framePtr, unsigned int minFrameDuration, unsigned int fadeDuration, int userIndex, bool purgeQueue);
-int speechPlayer_synthesize(speechPlayer_handle_t playerHandle, unsigned int sampleCount, sample* sampleBuf); 
+int speechPlayer_synthesize(speechPlayer_handle_t playerHandle, unsigned int sampleCount, ::sample* sampleBuf); 
 int speechPlayer_getLastIndex(speechPlayer_handle_t playerHandle);
 void speechPlayer_terminate(speechPlayer_handle_t playerHandle);
 
diff --git a/src/speechPlayer/src/speechPlayer.cpp b/src/speechPlayer/src/speechPlayer.cpp
index 30a970c4..c0ac52c0 100644
--- a/src/speechPlayer/src/speechPlayer.cpp
+++ b/src/speechPlayer/src/speechPlayer.cpp
@@ -38,7 +38,7 @@ void speechPlayer_queueFrame(speechPlayer_handle_t playerHandle, speechPlayer_fr
 	playerHandleInfo->frameManager->queueFrame(framePtr,minFrameDuration,fadeDuration,userIndex,purgeQueue);
 }
 
-int speechPlayer_synthesize(speechPlayer_handle_t playerHandle, unsigned int sampleCount, sample* sampleBuf) {
+int speechPlayer_synthesize(speechPlayer_handle_t playerHandle, unsigned int sampleCount, ::sample* sampleBuf) {
 	return ((speechPlayer_handleInfo_t*)playerHandle)->waveGenerator->generate(sampleCount,sampleBuf);
 }
 
diff --git a/src/speechPlayer/src/speechPlayer.h b/src/speechPlayer/src/speechPlayer.h
index d19d695e..4c0a8780 100644
--- a/src/speechPlayer/src/speechPlayer.h
+++ b/src/speechPlayer/src/speechPlayer.h
@@ -27,7 +27,7 @@ typedef void* speechPlayer_handle_t;
 
 speechPlayer_handle_t speechPlayer_initialize(int sampleRate);
 void speechPlayer_queueFrame(speechPlayer_handle_t playerHandle, speechPlayer_frame_t* framePtr, unsigned int minFrameDuration, unsigned int fadeDuration, int userIndex, bool purgeQueue);
-int speechPlayer_synthesize(speechPlayer_handle_t playerHandle, unsigned int sampleCount, sample* sampleBuf); 
+int speechPlayer_synthesize(speechPlayer_handle_t playerHandle, unsigned int sampleCount, ::sample* sampleBuf); 
 int speechPlayer_getLastIndex(speechPlayer_handle_t playerHandle);
 void speechPlayer_terminate(speechPlayer_handle_t playerHandle);
 
diff --git a/src/speechPlayer/src/speechWaveGenerator.cpp b/src/speechPlayer/src/speechWaveGenerator.cpp
index df483192..610ec3b3 100644
--- a/src/speechPlayer/src/speechWaveGenerator.cpp
+++ b/src/speechPlayer/src/speechWaveGenerator.cpp
@@ -194,7 +194,7 @@ class SpeechWaveGeneratorImpl: public SpeechWaveGenerator {
 	SpeechWaveGeneratorImpl(int sr): sampleRate(sr), voiceGenerator(sr), fricGenerator(), cascade(sr), parallel(sr), frameManager(NULL) {
 	}
 
-	unsigned int generate(const unsigned int sampleCount, sample* sampleBuf) {
+	unsigned int generate(const unsigned int sampleCount, ::sample* sampleBuf) {
 		if(!frameManager) return 0; 
 		double val=0;
 		for(unsigned int i=0;i<sampleCount;++i) {
diff --git a/src/speechPlayer/src/waveGenerator.h b/src/speechPlayer/src/waveGenerator.h
index f45de8cc..ea3b5e63 100644
--- a/src/speechPlayer/src/waveGenerator.h
+++ b/src/speechPlayer/src/waveGenerator.h
@@ -22,7 +22,7 @@ http://www.gnu.org/licenses/gpl.html
 
 class WaveGenerator {
 	public:
-	virtual unsigned int generate(const unsigned int bufSize, sample* buffer)=0;
+	virtual unsigned int generate(const unsigned int bufSize, ::sample* buffer)=0;
 };
 
 #endif
