diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 1c3832cf..baa96ba8 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -33,11 +33,19 @@ endmacro(compiled_test)
 find_program(SHELL bash)
 
 macro(shell_test _test_name)
-  add_test(
-    NAME ${_test_name}
-    COMMAND ${ESPEAK_RUN_ENV} ESPEAK_BIN=$<TARGET_FILE:espeak-ng-bin> ${SHELL} ${CMAKE_CURRENT_SOURCE_DIR}/${_test_name}.test
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/..
-  )
+  if(TARGET espeak-ng-bin)
+    add_test(
+      NAME ${_test_name}
+      COMMAND ${ESPEAK_RUN_ENV} ESPEAK_BIN=$<TARGET_FILE:espeak-ng-bin> ${SHELL} ${CMAKE_CURRENT_SOURCE_DIR}/${_test_name}.test
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/..
+    )
+  elseif(HOST_ESPEAK_NG)
+    add_test(
+      NAME ${_test_name}
+      COMMAND ${ESPEAK_RUN_ENV} ESPEAK_BIN=${HOST_ESPEAK_NG} ${SHELL} ${CMAKE_CURRENT_SOURCE_DIR}/${_test_name}.test
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/..
+    )
+  endif()
 endmacro(shell_test)
 
 compiled_test(api)
