diff --git a/cmake/onnxruntime_providers_dml.cmake b/cmake/onnxruntime_providers_dml.cmake
index e5ff19f133..1f35d46209 100644
--- a/cmake/onnxruntime_providers_dml.cmake
+++ b/cmake/onnxruntime_providers_dml.cmake
@@ -48,6 +48,11 @@
     else()
       add_dependencies(${target} RESTORE_PACKAGES)
-      target_link_libraries(${target} PRIVATE "${DML_PACKAGE_DIR}/bin/${onnxruntime_target_platform}-win/DirectML.lib")
-        target_compile_definitions(${target} PRIVATE DML_TARGET_VERSION_USE_LATEST)
+      target_link_libraries(
+          ${target}
+          PRIVATE
+            "$<BUILD_INTERFACE:${DML_PACKAGE_DIR}/bin/${onnxruntime_target_platform}-win/DirectML.lib>"
+            "$<INSTALL_INTERFACE:DirectML.lib>")
+      target_compile_definitions(${target} PRIVATE DML_TARGET_VERSION_USE_LATEST)
+      install(FILES "${DML_PACKAGE_DIR}/bin/${onnxruntime_target_platform}-win/DirectML.lib" TYPE LIB)
     endif()
   endfunction()
