diff --git a/onnxruntime/core/providers/xnnpack/tensor/resize.cc b/onnxruntime/core/providers/xnnpack/tensor/resize.cc
--- a/onnxruntime/core/providers/xnnpack/tensor/resize.cc
+++ b/onnxruntime/core/providers/xnnpack/tensor/resize.cc
@@ -66,6 +66,9 @@ bool Resize::IsOnnxNodeSupported(const NodeUnit& node_unit,
     // if not, it is not the C dim as a Resize will not change the number of channels.
     if (scale_tensor) {
       const Initializer scale_val(*scale_tensor, node_unit.ModelPath());
+      if (scale_val.size() < 4) {
+        break;  // empty or insufficient scales
+      }
       const auto scales = scale_val.DataAsSpan<float>();
       if (scales[1] != 1.0F) {
         break;
