diff --git a/onnxruntime/contrib_ops/cpu/bert/bias_gelu.cc b/onnxruntime/contrib_ops/cpu/bert/bias_gelu.cc
--- a/onnxruntime/contrib_ops/cpu/bert/bias_gelu.cc
+++ b/onnxruntime/contrib_ops/cpu/bert/bias_gelu.cc
@@ -23,6 +23,15 @@
     KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
     BiasGelu<float, false>);
 
+ONNX_OPERATOR_TYPED_KERNEL_EX(
+    BiasGelu,
+    kMSDomain,
+    1,
+    MLFloat16,
+    kCpuExecutionProvider,
+    KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<MLFloat16>()),
+    BiasGelu<MLFloat16, false>);
+
 // FastGelu uses approximation for Gelu. The formula is 0.5 * (1 + Tanh(x * (C * x * x + B))) * x.
 static constexpr float B = 0.7978845608028654f;    // sqrt(2.0 / M_PI)
 static constexpr float C = 0.035677408136300125f;  // 0.044715 * sqrt(2.0 / M_PI)
@@ -132,5 +142,70 @@
 // Instantiation for FastGelu
 template class BiasGelu<float, true>;
 
+template <>
+Status BiasGelu<MLFloat16, false>::Compute(OpKernelContext* context) const {
+  ORT_RETURN_IF_ERROR(bias_gelu_helper::CheckInputs(context));
+
+  const Tensor* input = context->Input<Tensor>(0);
+  const MLFloat16* input_data = input->Data<MLFloat16>();
+  int64_t elem_count = input->Shape().Size();
+
+  Tensor* output = context->Output(0, input->Shape());
+  MLFloat16* output_data = output->MutableData<MLFloat16>();
+
+  const Tensor* bias = context->Input<Tensor>(1);
+  const MLFloat16* bias_data = bias->Data<MLFloat16>();
+  int64_t bias_len = bias->Shape().Size();
+
+  AllocatorPtr alloc;
+  ORT_RETURN_IF_ERROR(context->GetTempSpaceAllocator(&alloc));
+
+  BufferUniquePtr input_f32_buf(alloc->Alloc(SafeInt<size_t>(sizeof(float)) * elem_count), BufferDeleter(alloc));
+  BufferUniquePtr bias_f32_buf(alloc->Alloc(SafeInt<size_t>(sizeof(float)) * bias_len), BufferDeleter(alloc));
+  BufferUniquePtr output_f32_buf(alloc->Alloc(SafeInt<size_t>(sizeof(float)) * elem_count), BufferDeleter(alloc));
+  BufferUniquePtr tmp_f32_buf(alloc->Alloc(SafeInt<size_t>(sizeof(float)) * elem_count), BufferDeleter(alloc));
+
+  float* input_f32 = static_cast<float*>(input_f32_buf.get());
+  float* bias_f32 = static_cast<float*>(bias_f32_buf.get());
+  float* output_f32 = static_cast<float*>(output_f32_buf.get());
+  float* tmp_f32 = static_cast<float*>(tmp_f32_buf.get());
+
+  for (int64_t i = 0; i < elem_count; i++) {
+    input_f32[i] = input_data[i].ToFloat();
+  }
+  for (int64_t i = 0; i < bias_len; i++) {
+    bias_f32[i] = bias_data[i].ToFloat();
+  }
+
+  int64_t task_count = elem_count / bias_len;
+
+  concurrency::ThreadPool::TryBatchParallelFor(
+      context->GetOperatorThreadPool(), static_cast<int32_t>(task_count),
+      [&](ptrdiff_t task_idx) {
+        const float* p_input = input_f32 + task_idx * bias_len;
+        float* p_output = output_f32 + task_idx * bias_len;
+        float* p_tmp = tmp_f32 + task_idx * bias_len;
+
+        for (int64_t i = 0; i < bias_len; i++) {
+          float value = p_input[i] + bias_f32[i];
+          p_output[i] = value * static_cast<float>(M_SQRT1_2);
+          p_tmp[i] = value * 0.5f;
+        }
+
+        MlasComputeErf(p_output, p_output, narrow<size_t>(bias_len));
+
+        for (int64_t i = 0; i < bias_len; i++) {
+          p_output[i] = p_tmp[i] * (p_output[i] + 1.0f);
+        }
+      },
+      0);
+
+  for (int64_t i = 0; i < elem_count; i++) {
+    output_data[i] = MLFloat16(output_f32[i]);
+  }
+
+  return Status::OK();
+}
+
 }  // namespace contrib
 }  // namespace onnxruntime
diff --git a/onnxruntime/contrib_ops/cpu/cpu_contrib_kernels.cc b/onnxruntime/contrib_ops/cpu/cpu_contrib_kernels.cc
--- a/onnxruntime/contrib_ops/cpu/cpu_contrib_kernels.cc
+++ b/onnxruntime/contrib_ops/cpu/cpu_contrib_kernels.cc
@@ -60,6 +60,7 @@
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, double, CDist);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, Gelu);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, BiasGelu);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, MLFloat16, BiasGelu);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, FastGelu);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, NGramRepeatBlock);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, BifurcationDetector);
@@ -343,6 +344,7 @@
       BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, float, CDist)>,
       BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, double, CDist)>,
       BuildKernelCreateInfo<ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, BiasGelu)>,
+      BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, MLFloat16, BiasGelu)>,
       BuildKernelCreateInfo<ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, Gelu)>,
       BuildKernelCreateInfo<ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, FastGelu)>,
       BuildKernelCreateInfo<ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, NGramRepeatBlock)>,
