--- a/onnxruntime/core/providers/xnnpack/tensor/resize.cc
+++ b/onnxruntime/core/providers/xnnpack/tensor/resize.cc
@@ -228,13 +228,13 @@
   auto out_h = output_dims_[1];
   auto out_w = output_dims_[2];
   if (op_type_ == OpComputeType::op_compute_type_fp32) {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_fp32, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_f32(out_h, out_w, flags, &p);
   } else if (op_type_ == OpComputeType::op_compute_type_fp16) {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_fp16, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_f16(out_h, out_w, flags, &p);
   } else if (op_type_ == OpComputeType::op_compute_type_qu8) {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_quint8, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_u8(out_h, out_w, flags, &p);
   } else {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_qint8, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_s8(out_h, out_w, flags, &p);
   }
 
   ORT_ENFORCE(xstatus == xnn_status_success, "xnn_create_resize_bilinear2d_nhwc_", OpTypeToString(op_type_), " failed. Status:",
@@ -261,10 +261,21 @@
   auto deallocator = [allocator](void* ptr) { allocator->aligned_deallocate(allocator->context, ptr); };
   std::unique_ptr<void, decltype(deallocator)> workspace(nullptr, deallocator);
 
-  auto reshape_fn = xnn_reshape_resize_bilinear2d_nhwc;
-
-  auto status = reshape_fn(op0_.get(), N, H, W, C, C, C,
-                           &workspace_size, threadpool);
+  auto status = xnn_status_invalid_state;
+
+  if (op_type_ == OpComputeType::op_compute_type_fp32) {
+    status = xnn_reshape_resize_bilinear2d_nhwc_f32(op0_.get(), N, H, W, C, C, C,
+                                                    &workspace_size, nullptr, threadpool);
+  } else if (op_type_ == OpComputeType::op_compute_type_fp16) {
+    status = xnn_reshape_resize_bilinear2d_nhwc_f16(op0_.get(), N, H, W, C, C, C,
+                                                    &workspace_size, nullptr, threadpool);
+  } else if (op_type_ == OpComputeType::op_compute_type_qu8) {
+    status = xnn_reshape_resize_bilinear2d_nhwc_u8(op0_.get(), N, H, W, C, C, C,
+                                                   &workspace_size, nullptr, threadpool);
+  } else {
+    status = xnn_reshape_resize_bilinear2d_nhwc_s8(op0_.get(), N, H, W, C, C, C,
+                                                   &workspace_size, nullptr, threadpool);
+  }
   if (status != xnn_status_success) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "xnn_reshape_resize_bilinear2d_nhwc_", OpTypeToString(op_type_),
                            " returned ", status);
@@ -273,17 +284,17 @@
   workspace.reset(allocator->aligned_allocate(allocator->context, XNN_ALLOCATION_ALIGNMENT, workspace_size));
 
   if (op_type_ == OpComputeType::op_compute_type_fp32) {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<float>(),
-                                              output->MutableData<float>());
+    status = xnn_setup_resize_bilinear2d_nhwc_f32(op0_.get(), workspace.get(), input->Data<float>(),
+                                                  output->MutableData<float>());
   } else if (op_type_ == OpComputeType::op_compute_type_fp16) {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<MLFloat16>(),
-                                              output->MutableData<MLFloat16>());
+    status = xnn_setup_resize_bilinear2d_nhwc_f16(op0_.get(), workspace.get(), input->Data<MLFloat16>(),
+                                                  output->MutableData<MLFloat16>());
   } else if (op_type_ == OpComputeType::op_compute_type_qu8) {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<uint8_t>(),
-                                              output->MutableData<uint8_t>());
+    status = xnn_setup_resize_bilinear2d_nhwc_u8(op0_.get(), workspace.get(), input->Data<uint8_t>(),
+                                                 output->MutableData<uint8_t>());
   } else {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<int8_t>(),
-                                              output->MutableData<int8_t>());
+    status = xnn_setup_resize_bilinear2d_nhwc_s8(op0_.get(), workspace.get(), input->Data<int8_t>(),
+                                                 output->MutableData<int8_t>());
   }
 
   if (status != xnn_status_success) {
--- a/onnxruntime/core/providers/xnnpack/tensor/resize.cc
+++ b/onnxruntime/core/providers/xnnpack/tensor/resize.cc
@@ -228,13 +228,13 @@
   auto out_h = output_dims_[1];
   auto out_w = output_dims_[2];
   if (op_type_ == OpComputeType::op_compute_type_fp32) {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_fp32, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_f32(out_h, out_w, flags, &p);
   } else if (op_type_ == OpComputeType::op_compute_type_fp16) {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_fp16, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_f16(out_h, out_w, flags, &p);
   } else if (op_type_ == OpComputeType::op_compute_type_qu8) {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_quint8, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_u8(out_h, out_w, flags, &p);
   } else {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_qint8, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_s8(out_h, out_w, flags, &p);
   }
 
   ORT_ENFORCE(xstatus == xnn_status_success, "xnn_create_resize_bilinear2d_nhwc_", OpTypeToString(op_type_), " failed. Status:",
@@ -261,10 +261,21 @@
   auto deallocator = [allocator](void* ptr) { allocator->aligned_deallocate(allocator->context, ptr); };
   std::unique_ptr<void, decltype(deallocator)> workspace(nullptr, deallocator);
 
-  auto reshape_fn = xnn_reshape_resize_bilinear2d_nhwc;
-
-  auto status = reshape_fn(op0_.get(), N, H, W, C, C, C,
-                           &workspace_size, threadpool);
+  auto status = xnn_status_invalid_state;
+
+  if (op_type_ == OpComputeType::op_compute_type_fp32) {
+    status = xnn_reshape_resize_bilinear2d_nhwc_f32(op0_.get(), N, H, W, C, C, C,
+                                                    &workspace_size, nullptr, threadpool);
+  } else if (op_type_ == OpComputeType::op_compute_type_fp16) {
+    status = xnn_reshape_resize_bilinear2d_nhwc_f16(op0_.get(), N, H, W, C, C, C,
+                                                    &workspace_size, nullptr, threadpool);
+  } else if (op_type_ == OpComputeType::op_compute_type_qu8) {
+    status = xnn_reshape_resize_bilinear2d_nhwc_u8(op0_.get(), N, H, W, C, C, C,
+                                                   &workspace_size, nullptr, threadpool);
+  } else {
+    status = xnn_reshape_resize_bilinear2d_nhwc_s8(op0_.get(), N, H, W, C, C, C,
+                                                   &workspace_size, nullptr, threadpool);
+  }
   if (status != xnn_status_success) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "xnn_reshape_resize_bilinear2d_nhwc_", OpTypeToString(op_type_),
                            " returned ", status);
@@ -273,17 +284,17 @@
   workspace.reset(allocator->aligned_allocate(allocator->context, XNN_ALLOCATION_ALIGNMENT, workspace_size));
 
   if (op_type_ == OpComputeType::op_compute_type_fp32) {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<float>(),
-                                              output->MutableData<float>());
+    status = xnn_setup_resize_bilinear2d_nhwc_f32(op0_.get(), workspace.get(), input->Data<float>(),
+                                                  output->MutableData<float>());
   } else if (op_type_ == OpComputeType::op_compute_type_fp16) {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<MLFloat16>(),
-                                              output->MutableData<MLFloat16>());
+    status = xnn_setup_resize_bilinear2d_nhwc_f16(op0_.get(), workspace.get(), input->Data<MLFloat16>(),
+                                                  output->MutableData<MLFloat16>());
   } else if (op_type_ == OpComputeType::op_compute_type_qu8) {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<uint8_t>(),
-                                              output->MutableData<uint8_t>());
+    status = xnn_setup_resize_bilinear2d_nhwc_u8(op0_.get(), workspace.get(), input->Data<uint8_t>(),
+                                                 output->MutableData<uint8_t>());
   } else {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<int8_t>(),
-                                              output->MutableData<int8_t>());
+    status = xnn_setup_resize_bilinear2d_nhwc_s8(op0_.get(), workspace.get(), input->Data<int8_t>(),
+                                                 output->MutableData<int8_t>());
   }
 
   if (status != xnn_status_success) {
--- a/onnxruntime/core/providers/xnnpack/tensor/resize.cc
+++ b/onnxruntime/core/providers/xnnpack/tensor/resize.cc
@@ -228,13 +228,13 @@
   auto out_h = output_dims_[1];
   auto out_w = output_dims_[2];
   if (op_type_ == OpComputeType::op_compute_type_fp32) {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_fp32, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_f32(out_h, out_w, flags, &p);
   } else if (op_type_ == OpComputeType::op_compute_type_fp16) {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_fp16, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_f16(out_h, out_w, flags, &p);
   } else if (op_type_ == OpComputeType::op_compute_type_qu8) {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_quint8, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_u8(out_h, out_w, flags, &p);
   } else {
-    xstatus = xnn_create_resize_bilinear2d_nhwc(xnn_datatype_qint8, out_h, out_w, flags, &p);
+    xstatus = xnn_create_resize_bilinear2d_nhwc_s8(out_h, out_w, flags, &p);
   }
 
   ORT_ENFORCE(xstatus == xnn_status_success, "xnn_create_resize_bilinear2d_nhwc_", OpTypeToString(op_type_), " failed. Status:",
@@ -261,10 +261,21 @@
   auto deallocator = [allocator](void* ptr) { allocator->aligned_deallocate(allocator->context, ptr); };
   std::unique_ptr<void, decltype(deallocator)> workspace(nullptr, deallocator);
 
-  auto reshape_fn = xnn_reshape_resize_bilinear2d_nhwc;
-
-  auto status = reshape_fn(op0_.get(), N, H, W, C, C, C,
-                           &workspace_size, threadpool);
+  auto status = xnn_status_invalid_state;
+
+  if (op_type_ == OpComputeType::op_compute_type_fp32) {
+    status = xnn_reshape_resize_bilinear2d_nhwc_f32(op0_.get(), N, H, W, C, C, C,
+                                                    &workspace_size, nullptr, threadpool);
+  } else if (op_type_ == OpComputeType::op_compute_type_fp16) {
+    status = xnn_reshape_resize_bilinear2d_nhwc_f16(op0_.get(), N, H, W, C, C, C,
+                                                    &workspace_size, nullptr, threadpool);
+  } else if (op_type_ == OpComputeType::op_compute_type_qu8) {
+    status = xnn_reshape_resize_bilinear2d_nhwc_u8(op0_.get(), N, H, W, C, C, C,
+                                                   &workspace_size, nullptr, threadpool);
+  } else {
+    status = xnn_reshape_resize_bilinear2d_nhwc_s8(op0_.get(), N, H, W, C, C, C,
+                                                   &workspace_size, nullptr, threadpool);
+  }
   if (status != xnn_status_success) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "xnn_reshape_resize_bilinear2d_nhwc_", OpTypeToString(op_type_),
                            " returned ", status);
@@ -273,17 +284,17 @@
   workspace.reset(allocator->aligned_allocate(allocator->context, XNN_ALLOCATION_ALIGNMENT, workspace_size));
 
   if (op_type_ == OpComputeType::op_compute_type_fp32) {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<float>(),
-                                              output->MutableData<float>());
+    status = xnn_setup_resize_bilinear2d_nhwc_f32(op0_.get(), workspace.get(), input->Data<float>(),
+                                                  output->MutableData<float>());
   } else if (op_type_ == OpComputeType::op_compute_type_fp16) {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<MLFloat16>(),
-                                              output->MutableData<MLFloat16>());
+    status = xnn_setup_resize_bilinear2d_nhwc_f16(op0_.get(), workspace.get(), input->Data<MLFloat16>(),
+                                                  output->MutableData<MLFloat16>());
   } else if (op_type_ == OpComputeType::op_compute_type_qu8) {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<uint8_t>(),
-                                              output->MutableData<uint8_t>());
+    status = xnn_setup_resize_bilinear2d_nhwc_u8(op0_.get(), workspace.get(), input->Data<uint8_t>(),
+                                                 output->MutableData<uint8_t>());
   } else {
-    status = xnn_setup_resize_bilinear2d_nhwc(op0_.get(), workspace.get(), input->Data<int8_t>(),
-                                              output->MutableData<int8_t>());
+    status = xnn_setup_resize_bilinear2d_nhwc_s8(op0_.get(), workspace.get(), input->Data<int8_t>(),
+                                                 output->MutableData<int8_t>());
   }
 
   if (status != xnn_status_success) {
