diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8e1f55..6d09bb0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,7 @@ elseif(NOT APPLE)
 endif()
 
 add_library(
-    piper_phonemize SHARED
+    piper_phonemize STATIC
     src/phonemize.cpp
     src/phoneme_ids.cpp
     src/tashkeel.cpp
@@ -31,7 +31,6 @@ add_library(
 
 set_target_properties(piper_phonemize PROPERTIES
     VERSION ${PROJECT_VERSION}
-    SOVERSION ${PROJECT_VERSION_MAJOR}
 )
 
 # ---- onnxruntime ---
