diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5917a56..ed8e8cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,13 @@ elseif(NOT APPLE)
   string(APPEND CMAKE_C_FLAGS " -Wall -Wextra")
 endif()
 
-add_executable(piper src/cpp/main.cpp src/cpp/piper.cpp)
+if(NOT IOS)
+    add_executable(piper src/cpp/main.cpp src/cpp/piper.cpp)
+else()
+    # For iOS builds, we don't build the main executable since it can't run on iOS
+    message(STATUS "iOS build detected: Skipping piper executable build")
+endif()
+
 add_executable(test_piper src/cpp/test.cpp src/cpp/piper.cpp)
 
 # ---- fmt ---
@@ -66,38 +72,40 @@ target_link_libraries(piper_core
     ${PIPER_PHONEMIZE_LIB}
 )
 
-# ---- Declare executable ----
-target_include_directories(piper PRIVATE 
-  ${ESPEAK_NG_INCLUDE_DIR}
-  ${PIPER_PHONEMIZE_INCLUDE_DIR}
-)
-
-if((NOT MSVC) AND (NOT APPLE))
-  # Linux flags
-  target_link_libraries(piper -static-libgcc -static-libstdc++)
-  # Don't link pthread on Android as it's included in the C library
-  if(NOT ANDROID)
-    set(PIPER_EXTRA_LIBRARIES "pthread")
-  endif()
+# ---- Configure executable ----
+if(NOT IOS)
+    target_include_directories(piper PRIVATE 
+      ${ESPEAK_NG_INCLUDE_DIR}
+      ${PIPER_PHONEMIZE_INCLUDE_DIR}
+    )
+
+    if((NOT MSVC) AND (NOT APPLE))
+      # Linux flags
+      target_link_libraries(piper -static-libgcc -static-libstdc++)
+      # Don't link pthread on Android as it's included in the C library
+      if(NOT ANDROID)
+        set(PIPER_EXTRA_LIBRARIES "pthread")
+      endif()
+    endif()
+
+    target_link_libraries(piper
+      fmt::fmt
+      spdlog::spdlog
+      ${ESPEAK_NG_LIB}
+      ${UCD_LIB}
+      ${SPEECHPLAYER_LIB}
+      ${PIPER_PHONEMIZE_LIB}
+      onnxruntime::onnxruntime_static
+    )
+
+    # Link pthread only if defined (not on Android)
+    if(DEFINED PIPER_EXTRA_LIBRARIES)
+      target_link_libraries(piper ${PIPER_EXTRA_LIBRARIES})
+    endif()
+
+    target_compile_definitions(piper PUBLIC _PIPER_VERSION=${piper_version})
 endif()
 
-target_link_libraries(piper
-  fmt::fmt
-  spdlog::spdlog
-  ${ESPEAK_NG_LIB}
-  ${UCD_LIB}
-  ${SPEECHPLAYER_LIB}
-  ${PIPER_PHONEMIZE_LIB}
-  onnxruntime::onnxruntime_static
-)
-
-# Link pthread only if defined (not on Android)
-if(DEFINED PIPER_EXTRA_LIBRARIES)
-  target_link_libraries(piper ${PIPER_EXTRA_LIBRARIES})
-endif()
-
-target_compile_definitions(piper PUBLIC _PIPER_VERSION=${piper_version})
-
 # ---- Declare test ----
 include(CTest)
 enable_testing()
@@ -125,10 +133,17 @@ target_link_libraries(test_piper PUBLIC
 
 # ---- Declare install targets ----
 
-install(
-  TARGETS piper piper_core
-  RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}
-  ARCHIVE DESTINATION lib)
+if(NOT IOS)
+    install(
+      TARGETS piper piper_core
+      RUNTIME DESTINATION bin
+      BUNDLE DESTINATION bin
+      ARCHIVE DESTINATION lib)
+else()
+    install(
+      TARGETS piper_core
+      ARCHIVE DESTINATION lib)
+endif()
 
 # Install piper_core headers
 install(
