diff --git a/cmake/ggml-config.cmake.in b/cmake/ggml-config.cmake.in
index 91c9d5c..b7d8ec2 100644
--- a/cmake/ggml-config.cmake.in
+++ b/cmake/ggml-config.cmake.in
@@ -124,6 +124,11 @@ if(NOT TARGET ggml::ggml)
         PROPERTIES
             IMPORTED_LOCATION "${GGML_BASE_LIBRARY}")
 
+    if(GGML_MAX_NAME)
+        set_property(TARGET ggml::ggml-base APPEND PROPERTY
+            INTERFACE_COMPILE_DEFINITIONS "GGML_MAX_NAME=${GGML_MAX_NAME}")
+    endif()
+
     set(_ggml_all_targets "")
     if (NOT GGML_BACKEND_DL)
         foreach(_ggml_backend ${GGML_AVAILABLE_BACKENDS})
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2650237..d0987d1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -221,6 +221,10 @@ if (GGML_SCHED_NO_REALLOC)
     target_compile_definitions(ggml-base PUBLIC GGML_SCHED_NO_REALLOC)
 endif()
 
+if (DEFINED GGML_MAX_NAME)
+    target_compile_definitions(ggml-base PUBLIC GGML_MAX_NAME=${GGML_MAX_NAME})
+endif()
+
 add_library(ggml
             ggml-backend-dl.cpp
             ggml-backend-reg.cpp)
