diff --git a/ggml/CMakeLists.txt b/ggml/CMakeLists.txt
index fd3cceb..d072fe6 100644
--- a/ggml/CMakeLists.txt
+++ b/ggml/CMakeLists.txt
@@ -58,7 +58,9 @@ else()
     set(GGML_BLAS_VENDOR_DEFAULT "Generic")
 endif()
 
-if (CMAKE_CROSSCOMPILING OR DEFINED ENV{SOURCE_DATE_EPOCH})
+if (CMAKE_CROSSCOMPILING OR DEFINED ENV{SOURCE_DATE_EPOCH} OR 
+    (APPLE AND CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "arm64" AND 
+     CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64"))
     message(STATUS "Setting GGML_NATIVE_DEFAULT to OFF")
     set(GGML_NATIVE_DEFAULT OFF)
 else()
