diff --git a/cmake/whisper-config.cmake.in b/cmake/whisper-config.cmake.in
index 678ff9f0..16162504 100644
--- a/cmake/whisper-config.cmake.in
+++ b/cmake/whisper-config.cmake.in
@@ -13,7 +13,9 @@ set(GGML_ACCELERATE @GGML_ACCELERATE@)
 
 include(CMakeFindDependencyMacro)
 
-find_dependency(Threads REQUIRED)
+if(NOT WIN32)
+    find_dependency(Threads REQUIRED)
+endif()
 
 if (APPLE AND GGML_ACCELERATE)
     find_library(ACCELERATE_FRAMEWORK Accelerate REQUIRED)
